watertap/property_models/activated_sludge/new_modified_asm2d_properties.py

```diff
@@ -0,0 +1,382 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+"""
+Thermophysical property package to be used in conjunction with modified ASM2d reactions.
+
+Reference:
+X. Flores-Alsina, K. Solon, C.K. Mbamba, S. Tait, K.V. Gernaey, U. Jeppsson, D.J. Batstone,
+Modelling phosphorus (P), sulfur (S) and iron (Fe) interactions fordynamic simulations of anaerobic digestion processes,
+Water Research. 95 (2016) 370-382. https://www.sciencedirect.com/science/article/pii/S0043135416301397
+
+"""
+
+# Import Pyomo libraries
+import pyomo.environ as pyo
+
+from enum import Enum, auto
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialFlowBasis,
+    PhysicalParameterBlock,
+    StateBlockData,
+    StateBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    LiquidPhase,
+    Solute,
+    Solvent,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import fix_state_vars, revert_state_vars
+import idaes.logger as idaeslog
+
+from pyomo.common.config import ConfigValue, In
+
+# Some more information about this module
+__author__ = "Marcus Holly"
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+class DecaySwitch(Enum):
+    on = auto()
+    off = auto()
+
+
+@declare_process_block_class("NewASM2dParameterBlock")
+class NewASM2dParameterData(PhysicalParameterBlock):
+    """
+    Property Parameter Block Class
+    """
+
+    CONFIG = PhysicalParameterBlock.CONFIG()
+
+    CONFIG.declare(
+        "decay_switch",
```

Comment:
Where does this get utilized in the model?

Reply:
A few of the rate expressions change depending on whether this is on or off. It is used in R9, R15, R16, R17, and R19.

Reply:
@MarcusHolly Actually, I just searched for the config option and didn't see it in the rxn package due to the file being truncated by Github. Since you mention it,  it does make sense for it to be a config option for the rxn package. Are there any drawbacks to making the change? At first glance it seems like it doesn't make a difference functionally, but it would make more sense logically.